sh_scrapy/crawl.py

```diff
@@ -139,7 +144,25 @@ def get_distribution():
     d = get_distribution()
     if not d:
         raise ValueError(SCRAPY_SETTINGS_ENTRYPOINT_NOT_FOUND)
-    d.run_script(scriptname, {'__name__': '__main__'})
+    ns = {"__name__": "__main__"}
+    if has_importlib:
+        _run_script(d, scriptname, ns)
+    else:
+        d.run_script(scriptname, ns)
+
+
+def _run_script(dist, script_name, namespace):
+    # an importlib-based replacement for pkg_resources.NullProvider.run_script()
+    script = "scripts/" + script_name
+    source = dist.read_text(script)
+    if not source:
+        raise ValueError(
+            f"Script {script!r} not found in metadata at {dist._path!r}"
+        )
+    script_filename = dist._path.joinpath(script)
```

Comment:
I don't see a way to avoid accessing `_path`, though it's used only in the exception above and to get full file paths in tracebacks from the compiled code, which should be fine to truncate to just `script` as users won't access scripts on the Dash system anyway.